test/contrib/hsgp/test_approximation.py

```diff
@@ -65,6 +72,130 @@ def synthetic_two_dim_data() -> tuple[ArrayImpl, ArrayImpl]:
     return generate_synthetic_two_dim_data(**kwargs)
 
 
+@pytest.mark.parametrize(
+    argnames="x1, x2, length, ell",
+    argvalues=[
+        (jnp.array([[1.0]]), jnp.array([[0.0]]), jnp.array([1.0]), 5.0),
```

Comment:
Could you change those global arrays into numpy arrays `np.array(...)`? We don't want to trigger any global jax code because users might need to change floating precision or the default device platform.

Reply:
Got it. We had a bunch of examples of this pattern in the hsgp test suite. Switched them all over and added some handling where needed for numpy arrays in the source code https://github.com/pyro-ppl/numpyro/pull/1811/commits/c9c1457cad9c2e230ab7f149ffa2925d119b3667

Reply:
Tests are failing because a type OR operation. I'm not sure about the type hint to use... Maybe we need 

```python
from jaxlib.xla_extension import ArrayImpl
``` 
?

Reply:
python 3.9...

let me know how this looks to you https://github.com/pyro-ppl/numpyro/pull/1811/commits/56e35f5a521ac9d40810b699b27d1f977c703b91

Reply:
It is green ✅🙌

Reply:
I'm not a JAX typing expert but this looks very readable 😄